service/telemetry/telemetry.go

```diff
@@ -39,18 +43,29 @@ type Settings struct {
 }
 
 // New creates a new Telemetry from Config.
-func New(_ context.Context, set Settings, cfg Config) (*Telemetry, error) {
+func New(_ context.Context, set Settings, cfg Config, tracerProvider trace.TracerProvider) (*Telemetry, error) {
```

Comment:
Have you discussed these changes with @jpkrohling? 

Something feels a bit off to me here with how the service-level providers are being injected down through the call stack; it doesn't feel like it's on the right trajectory for code that could eventually be merged.

Reply:
We have an effort upstream to instrument the collector, including for tracing. I'm quite confident the solution upstream is going to be quite different than what is here, but again, @mar4uk has a better context when it comes to this effort.

Reply:
There is a proposal to make the collector observable https://github.com/open-telemetry/opentelemetry-collector/issues/7532 upstream. 
But PRs are not moving for quite some time: https://github.com/open-telemetry/opentelemetry-collector/pull/7644
So it is difficult to say when this proposal will be implemented and go live. Probably not very soon